### PR TITLE
feat:  add mdns

### PR DESCRIPTION
**Pull request by gocarlos**
Specify library name and version:  **mdns/master**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mdns/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "20200225":
```
Comment (uilianries) [inline recipes/mdns/all/conandata.yml]:
```suggestion
  "20200130":
```

> Latest commit 0f75def 27 days ago

File: recipes/mdns/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import os
+from conans import CMake, ConanFile, tools
+
+
+class MdnsConan(ConanFile):
+    name = "mdns"
+    license = "Public Domain or MIT"
```
Comment (uilianries) [inline recipes/mdns/all/conanfile.py]:
```suggestion
    license = "Unlicense"
```

https://github.com/mjansson/mdns/blob/master/LICENSE

File: recipes/mdns/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    homepage = "https://github.com/mjansson/mdns"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """Public domain mDNS/DNS-SD library in C"""
+    topics = ("mdns")
```
Comment (uilianries) [inline recipes/mdns/all/conanfile.py]:
```suggestion
    topics = ("conan", "mdns", "dns", "dns-sd")
```

File: recipes/mdns/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import os
+from conans import CMake, ConanFile, tools
+
+
+class MdnsConan(ConanFile):
+    name = "mdns"
+    license = "Public Domain or MIT"
+    homepage = "https://github.com/mjansson/mdns"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """Public domain mDNS/DNS-SD library in C"""
```
Comment (uilianries) [inline recipes/mdns/all/conanfile.py]:
```suggestion
    description = "Public domain mDNS/DNS-SD library in C"
```

File: recipes/mdns/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                self.conan_data["sources"][self.version]["url"]).split(".")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*.h", dst="include", src=self._source_subfolder)
+
+    def package_info(self):
+        if self.settings.os == "Windows":
+            self.cpp_info.libs = ["iphlpapi", "ws2_32"]
```
Comment (uilianries) [inline recipes/mdns/all/conanfile.py]:
```suggestion
            self.cpp_info.system_libs = ["iphlpapi", "ws2_32"]
```

File: recipes/mdns/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        if str(self.settings.os) in ["Linux", "Android"]:
+            self.cpp_info.libs.append('pthread')
```
Comment (uilianries) [inline recipes/mdns/all/conanfile.py]:
```suggestion
            self.cpp_info.system_libs.append('pthread')
```

File: recipes/mdns/config.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+---
+versions:
+  "20200225":
```
Comment (uilianries) [inline recipes/mdns/config.yml]:
```suggestion
  "20200130":
```

File: recipes/mdns/all/test_package/test_package.c [hunk 8]
```diff
@@ -0,0 +1,363 @@
+
```
Comment (uilianries) [inline recipes/mdns/all/test_package/test_package.c]: could you please keep it shorter? You can use only a method (e.g. mdns_socket_open_ipv4) and remove all those user arguments.

**Comment by gocarlos** [comment]
@danimtb please review this

**Comment by uilianries** [comment]
I gonna restart it, because the last builds failed on CI.

**Pull request merged**